TraitsExpanded/Interfaces/TraitSet.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections.Generic;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace TraitsExpanded
+{
+    public abstract class TraitSet
+    {
+        public virtual List<ITrait> Traits { get; set; } = new List<ITrait>();
+        public virtual string ID { get; set; }
+        public virtual int CurrentTraitIndex { get; set; } = 0;
+        public virtual NPCRestrictionEnum NPCRestriction { get; set; } = NPCRestrictionEnum.NONE;
+        public virtual bool IsActive { get; set; } = false;
```

Comment:
is `IsActive` distinct from `CurrentTraidIndex == 0` ?

Reply:
So Active is a boolean to check whether the current TraitSystem is active on a character, CurrentTraitIndex is just the index of the trait that is currently active

Reply:
I see, because in the way you've done the indexes, 0 has no special meaning. we could say -1 is the sentinel value we choose for "traitset inactive" and IsActive could just be defined as `public bool IsActive => CurrentTraitIndex >= 0`